src/dialects/postgres/query-generator.js

```diff
@@ -869,38 +900,170 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
   /**
    * Generates an SQL query that returns all foreign keys of a table.
    *
-   * @param  {string} tableName  The name of the table.
-   * @returns {string}            The generated sql query.
+   * @param   {string|object} tableArg  string: Name of table
+   *                                    object: { tableName: Name of table, schema: name of schema }
+   * @param   {object}        options   Pass to extractTableDetails
+   * @returns {string}                  The generated sql query.
    * @private
    */
-  getForeignKeysQuery(tableName) {
-    return 'SELECT conname as constraint_name, pg_catalog.pg_get_constraintdef(r.oid, true) as condef FROM pg_catalog.pg_constraint r '
-      + `WHERE r.conrelid = (SELECT oid FROM pg_class WHERE relname = '${tableName}' LIMIT 1) AND r.contype = 'f' ORDER BY 1;`;
-  }
+  getForeignKeysQuery(tableArg, options) {
```

Comment:
IMO this function should not need a second argument, no need for `extractTableDetails`.

Instead, it should have the following signature:

```ts
getForeignKeysQuery(tableName: TableName): string;
```

`TableName` can be a string, or an object which includes the `tableName` and `schema` properties

If you want to filter by column too, pass columns in the option bag (not in the table name):

```ts
type GetForeignsKeyQueryOptions = {
  columns?: string[]
}

getForeignKeysQuery(tableName: TableName, options?: GetForeignsKeyQueryOptions): string;
```